Library/Homebrew/extend/os/linux/hardware/cpu.rb

```diff
@@ -20,12 +20,19 @@ def cpuinfo
       end
 
       def type
-        @type ||= if cpuinfo =~ /Intel|AMD/
-          :intel
-        elsif cpuinfo =~ /ARM|Marvell/
-          :arm
-        else
-          :dunno
+        @type ||= begin
+          uname = Utils.popen_read("uname", "-m").chomp
+          if uname == "x86_64"
+            :intel
+          elsif uname.start_with? "arm"
+            :arm
+          elsif cpuinfo =~ /Intel|AMD/
+            :intel
+          elsif cpuinfo =~ /ARM|Marvell/
+            :arm
```

Comment:
I'm not a fan of the redundancy. I did imply that's what I wanted. Sorry for that.

Let's delete lines 29–32 until we find a situation where they're necessary. What does `uname -m` return on a 32-bit x86 system? As I recall it matches `/86$/`, like `i386`, `i586` et c.
I oddly can't find the original issue for the `Marvel` ARM system. I'm fine ignoring that for now.
Let's use
```ruby
case Utils.popen_read("uname", "-m").chomp
```